core/modules/catalog/test/unit/helpers/optionLabel.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { optionLabel } from '@vue-storefront/core/modules/catalog/helpers/optionLabel';
+
+describe('optionLabel', () => {
+  it('returns proper opt.label and set it in state', () => {
+    const mockedData = {
+      state: {
+        attribute: {
+          labels: {
+            color: {}
+          },
+          list_by_code: {
+            color: {
+              options: [
+                {
+                  label: 'Orange',
+                  sort_order: 7,
+                  swatch: {
+                    type: 1,
+                    value: '#eb6703'
+                  },
+                  value: '5483'
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+    const label = optionLabel(mockedData.state.attribute, { attributeKey: 'color', optionId: 5483 })
+    const result = mockedData.state.attribute['labels'];
+    const expectedLabels = {
+      color: {
+        5483: 'Orange'
+      }
+    }
+    expect(label).toBe('Orange');
+    expect(result).toMatchObject(expectedLabels)
+  })
+
+  it('returns cached label', () => {
+    const mockedData = {
+      state: {
+        attribute: {
+          labels: {
+            color: {
+              5483: 'Orange'
+            }
+          }
+        }
+      }
+    }
+    optionLabel(mockedData.state.attribute, { attributeKey: 'color', optionId: 5483 })
+
+    const result = mockedData.state.attribute.labels.color['5483']
+    const expectedResult = 'Orange'
+
+    expect(result).toBe(expectedResult)
+  })
+
+  it('returns optionId if options.label doesn\'t exist', () => {
+    const mockedData = {
+      state: {
+        attribute: {
+          list_by_code: {
+            color: {
+              options: [
+                {
+                  sort_order: 7,
+                  swatch: {
+                    type: 1,
+                    value: '#eb6703'
+                  },
+                  value: '5483'
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+    const label = optionLabel(mockedData.state.attribute, { attributeKey: 'color', optionId: 5483 })
+
+    expect(label).toBe(5483);
+  })
+
+  it('returns optionId if attribute doesn\'t exist', () => {
+    const mockedData = {
+      state: {
+        attribute: {
+          list_by_code: {}
+        }
+      }
+    }
+    const label = optionLabel(mockedData.state.attribute, { attributeKey: 'color', optionId: 5483 })
+
+    expect(label).toBe(5483);
+  })
+
+  it('returns label when searchBy has other value than default', () => {
```

Comment:
Very good test. But it's description might be more precise. Maybe just `searchBy` works properly?